conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -940,38 +940,101 @@ public Boolean visitUnknown(String unknownType) {
+                }
+                return value.getItemType().accept(new DefaultTypeVisitor<>() {
+                    @Override
+                    public CollectionType visitDefault() {
+                        return new CollectionType(
+                                ConjureCollectionType.LIST,
+                                ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                    }
+
+                    @Override
+                    public CollectionType visitPrimitive(PrimitiveType primitiveType) {
+                        return primitiveType.accept(new DefaultPrimitiveTypeVisitor<>() {
+
+                            @Override
+                            public CollectionType visitDefault() {
+                                return new CollectionType(
+                                        ConjureCollectionType.LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+
+                            @Override
+                            public CollectionType visitDouble() {
+                                return new CollectionType(
+                                        ConjureCollectionType.DOUBLE_ARRAY_LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+
+                            @Override
+                            public CollectionType visitInteger() {
+                                return new CollectionType(
+                                        ConjureCollectionType.INTEGER_ARRAY_LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+
+                            @Override
+                            public CollectionType visitBoolean() {
+                                return new CollectionType(
+                                        ConjureCollectionType.BOOLEAN_ARRAY_LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+
+                            @Override
+                            public CollectionType visitSafelong() {
+                                return new CollectionType(
+                                        ConjureCollectionType.SAFE_LONG_ARRAY_LIST,
+                                        ConjureCollectionNullHandlingMode.NON_NULL_COLLECTION_FACTORY);
+                            }
+                        });
+                    }
+                });
             }
 
             @Override
             public CollectionType visitSet(SetType _value) {
-                return new CollectionType(ConjureCollectionType.SET, options.nonNullCollections());
+                if (options.nonNullCollections()) {
```

Comment:
Let's add a new temporary "experimental" feature flag to opt into codegen using the new methods, that way we can allow dependencies to roll out for a bit before folks start to use them. After a couple weeks we can remove the flag and make it the default, that way rollout shouldn't break folks who violate jar dependencies 😓

Reply:
good idea 4ef8ad246583c75959a4ee27f5c2cfca32fdf881